add docs for if-diff (#74)
Green-Software-Foundation/if-docs: docs/reference/cli.md

# Command line tool 

> # Command line tool 

Reviewer: redundant headline

Author: ```suggestion
```

A core feature of the Impact Framework is the `ie` command line tool (CLI). This is how you trigger Impact Framework to execute a certain manifest file. 

We also provide several other command line tools that work in concert with `ie` to enable flows such as comparing, re-executing and verifying IF output files.

> This page includes rfeerence documentation for the CLI tools, including the various commands and flags each tool exposes. 

Reviewer: rfeerence

Author: ```suggestion
This page includes reference documentation for the CLI tools, including the various commands and flags each tool exposes. 
```

## or using alias
ie -h
```

## `if-diff`

The `if-diff` command line tool allows you to determine whether two manifest or output files are the same, and if not, how they differ.

`if-diff` needs two files to compare - a `source` and a `target`. The `source` file is considered to be the "true" file that another file, the `target`, is compared against. Note that for most purposes, it doesn't matter which file is assigned as `source` or `target` - the important thing is that `if-diff` receives two files. Both files should be `yaml` files. They are expected to be IF output files, meaning they contain all the required fields of a `manifest` plus the IF-generated `output` and `execution` blocks.

`if-diff` is run as follows:

```sh
if-diff --source file-1.yml --target file2.yml
```

> You can also pipe the oputputs from `ie` directly into `if-diff`. This means you only provide *one* file to `if-diff` and the other comes from a new `ie` run configured to send its output data to the console via `stdout`. This is an important feature because it allows you to receive an output file and verify that it was computed correctly and not tampered with post-execution. For example, if someone provides you with an output file, you can strip out the `outputs` section and re-run it with `ie`, piping the outputs straight to `if-diff` to compare against the original you received. 

Reviewer: oputputs

Author: ```suggestion
You can also pipe the outputs from `ie` directly into `if-diff`. This means you only provide *one* file to `if-diff` and the other comes from a new `ie` run configured to send its output data to the console via `stdout`. This is an important feature because it allows you to receive an output file and verify that it was computed correctly and not tampered with post-execution. For example, if someone provides you with an output file, you can strip out the `outputs` section and re-run it with `ie`, piping the outputs straight to `if-diff` to compare against the original you received. 
```

| Difference identified                                    | Report or ignore? | Note                                                                                       |
| -------------------------------------------------------- | ----------------- | ------------------------------------------------------------------------------------------ |
| trees contain different number of nodes                  | report            | works the same regardless whether `source` or `target` has more nodes                      |
| nodes in tree have different names                       | report            | There should be no named nodes existing in one file that aren't also in the other          |
| nodes in tree contain non-identical fields and/or values | report            | the data inside each tree component should contain identical keys/values                   |

> | keys and values in context field are non- identical      | report            | the same fields should exist in the `context` sectiona dn their values should be identical |

Reviewer: sectiona dn

Author: ```suggestion
| keys and values in context field are non- identical      | report            | the same fields should exist in the `context` section and their values should be identical |
```
